### PR TITLE
Global theme

### PR DESCRIPTION
**Pull request by noszczynski**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <meta name="theme-color" content="#ffffff" />
     <meta
       name="description"
-      content="Web games store by Bortacki Tomasz, Adam Noszczyński and Michał Sroka"
+      content="Web games store by Bortacki Tomasz and Adam Noszczyński"
```
Comment (noszczynski) [inline public/index.html:9]: Michał currently is outside of the project

File: src/api/api.js [hunk 2]
```diff
@@ -5,10 +5,10 @@ const api = {
...
 // Getting games from api
-const apiGames = () => {
+const getGames = () => {
   return fetch(api.games).then((response) => {
     return response.json();
   });
 };
 
-export { apiUrl, api, apiGames };
+export { apiUrl, api, getGames };
```
Comment (noszczynski) [inline src/api/api.js:14]: IMO getGames is more clear

File: src/components/Input/Input.js [hunk 3]
```diff
@@ -1,29 +1,41 @@
...
   width: 100%;
-  font-size: 1rem;
-  font-weight: 400;
-  font-family: "Roboto", sans-serif;
-  border-radius: 0.25rem;
-  color: ${colors.gray300};
+  font-size: 1rem; // TODO
+  font-weight: 400; // TODO
+  font-family: "Roboto", sans-serif; // TODO
+  border-radius: 0.25rem; // TODO
```
Comment (noszczynski) [inline src/components/Input/Input.js:14]: this TODOs lines are ready for change into theme variable, but I have no time for this now

File: src/components/Input/Input.js [hunk 4]
```diff
@@ -1,29 +1,41 @@
...
     outline-width: 0;
-    background-color: ${colors.light};
+    background-color: ${({ theme }) => theme.colors.primaryBackgroundColor};
   }
 `;
 
-const Input = (props) => {
-  return <Text {...props} />;
+const Label = styled.label`
+  //
```
Comment (noszczynski) [inline src/components/Input/Input.js:25]: actually no place for this :p

Comment (noszczynski) [reply src/components/Input/Input.js:25]: In the next step You should styling this

File: src/components/Layout/Layout.js [hunk 5]
```diff
@@ -1,49 +1,136 @@
...
+      <GlobalStyle />
+      <Wrapper>
+        <SideMenu />
+        <Content clearTopPadding={removeTopPadding}>
+          <TopBar
+            items={data}
+            setter={setFilteredData}
+            setTheme={() => setCurrentTheme(handleSetTheme())}
+          />
+          <ContentWrapper>
```
Comment (noszczynski) [inline src/components/Layout/Layout.js:111]: Maybe separated component?

File: src/components/Layout/Layout.js [hunk 6]
```diff
@@ -1,49 +1,136 @@
...
+  padding: ${({ clearTopPadding }) => (clearTopPadding ? 0 : 88)}px 0
+    ${({ theme }) => theme.sizes.padding.lite};
+  grid-area: content;
+  background-color: ${({ theme }) => theme.colors.secondaryBackgroundColor};
+  transition: ${({ theme }) => theme.transitions.changeTheme};
+`;
+
+const ContentWrapper = styled.div`
+  padding: 0 ${({ theme }) => theme.sizes.padding.lite};
 `;
```
Comment (noszczynski) [inline src/components/Layout/Layout.js:38]: I think we should move it into separated file

File: src/components/Layout/TopBar.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  position: fixed;
+  top: 0;
+  right: 0;
+  left: calc(250px + 1rem);
+  z-index: 100;
+  padding: ${({ theme }) => `1.5rem ${theme.sizes.padding.lite} 1.5rem 0`};
+  background-color: ${({ theme }) => theme.colors.transparent};
+`;
+
+const RightWrapper = styled.div`
```
Comment (noszczynski) [inline src/components/Layout/TopBar.js:21]: actually I had no idea what name give him

File: src/components/Logo/Logo.js [hunk 8]
```diff
@@ -1,25 +1,36 @@
...
 `;
 
 const Logo = () => {
   return (
     <LogoContainer>
       <Link href={"/"}>
-        <span>GAMES STORE</span>
+        <p>
+          GAMES <span>STORE</span>
+        </p>
```
Comment (noszczynski) [inline src/components/Logo/Logo.js:33]: This should be any header like `h1` or `h2`

File: src/components/ProfileMenu/ProfileMenu.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+`;
+
+const ProfileMenu = () => {
+  return (
+    <Wrapper>
+      <Label>Nickname</Label>
+      <Photo>
+        <img
+          src={
+            "https://upload.wikimedia.org/wikipedia/commons/thumb/7/7e/Circle-icons-profile.svg/1200px-Circle-icons-profile.svg.png"
```
Comment (noszczynski) [inline src/components/ProfileMenu/ProfileMenu.js:42]: just mock

File: src/components/SideMenu/SideMenu.js [hunk 10]
```diff
@@ -23,18 +14,20 @@ const Aside = styled.aside`
...
-    store: { label: "Store", link: "/store", icon: <Store /> },
     games: { label: "Games", link: "/games", icon: <List /> },
-    favorite: { label: "Favorite", link: "/favorite", icon: <Favorite /> },
+    search: { label: "Search", link: "/search", icon: <Search /> },
+    profile: { label: "Profile", link: "/profile", icon: <Person /> },
+    // friend: { label: "Friend", link: "/friend", icon: <PersonAdd /> },
+    // messages: { label: "Messages", link: "/messages", icon: <ChatBubble /> },
+    // channels: { label: "Channels", link: "/channels", icon: <YouTube /> },
+    // store: { label: "Store", link: "/store", icon: <Store /> },
+    // favorite: { label: "Favorite", link: "/favorite", icon: <Favorite /> },
```
Comment (noszczynski) [inline src/components/SideMenu/SideMenu.js:30]: Someday time will come

File: src/profileReducer.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+export const reducer = (state, action) => {
+  const { type, target } = action;
+
+  switch (type) {
+    case PROFILE_ACTION_TYPES.CHANGE: {
+      const fieldName = target.getAttribute("name");
+      const { value } = target;
+
+      // TODO below switch convert into sth universal
```
Comment (noszczynski) [inline src/profileReducer.js:21]: like `TODO` said, You have take this in free time

File: src/utils/variables.js [hunk 12]
```diff
@@ -1,12 +1,21 @@
 import colors from "./colors";
 
 const shadows = {
-  first: `0 20px 50px ${colors.transparentDark}`,
```
Comment (noszczynski) [inline src/utils/variables.js:4]: Probably I wasn't changed this in the code :-1:

File: src/views/Profile.js [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+  align-items: center;
+`;
+
+const ProfileForm = styled.div`
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(320px, 1fr));
+  gap: ${({ theme }) => theme.sizes.padding.lite};
+`;
+
+const Profile = () => {
```
Comment (noszczynski) [inline src/views/Profile.js:19]: Someone should styling this

File: src/views/Profile.js [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+          />
+          <Input
+            type={"password"}
+            name={FIELD_NAMES.PASSWORD}
+            value={data.password}
+            placeholder={"pass"}
+            label={"Password"}
+            onChange={({ target }) =>
+              dispatch({ type: PROFILE_ACTION_TYPES.CHANGE, target })
+            }
```
Comment (noszczynski) [inline src/views/Profile.js:82]: I think confirm password field will be useful, but not now

File: src/views/SearchView.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+  const [searchTerm, setSearchTerm] = useState("");
+
+  return (
+    <Layout
+      title={"Search"}
+      searchTerm={searchTerm}
+      searchTermSetter={setSearchTerm}
+    >
+      <Wrapper>
+        <SearchedItem>search result mock</SearchedItem>
```
Comment (noszczynski) [inline src/views/SearchView.js:28]: ah, mocks...

File: src/components/Input/Input.js (hunk 3 above)
Comment (tomaszbortacki) [reply src/components/Input/Input.js:14]: Let's put this like that right now

**Pull request merged**